#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_use_wxt_for_gnuplot.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Enable the use of wxt as terminal for gnuplot via the GNUTERM environment variable
## DP: https://www.cae.wisc.edu/pipermail/help-octave/2007-July/004977.html

@DPATCH@
diff -urNad octave2.9-2.9.13~/scripts/plot/drawnow.m octave2.9-2.9.13/scripts/plot/drawnow.m
--- octave2.9-2.9.13~/scripts/plot/drawnow.m	2007-07-18 17:43:51.000000000 +0000
+++ octave2.9-2.9.13/scripts/plot/drawnow.m	2007-08-22 08:22:17.000000000 +0000
@@ -109,17 +109,19 @@
 
       ## Guess the terminal type.
       term = getenv ("GNUTERM");
-      if (isempty (term) && ! isempty (getenv ("DISPLAY")))
-        term = "x11";
-      elseif (! isunix ())
-        term = "windows";
-      else
-        term = "aqua";
-        ## This should really be checking for os x before setting
-        ## the terminal type to aqua, but nobody will notice because
-        ## every other unix will be using x11 and windows will be
-        ## using windows.  Those diehards still running octave from
-        ## a linux console know how to set the GNUTERM variable.
+      if (isempty (term))
+	if (! isempty (getenv ("DISPLAY")))
+          term = "x11";
+	elseif (! isunix ())
+          term = "windows";
+	else
+	  ## This should really be checking for os x before setting
+	  ## the terminal type to aqua, but nobody will notice because
+	  ## every other unix will be using x11 and windows will be
+	  ## using windows.  Those diehards still running octave from
+	  ## a linux console know how to set the GNUTERM variable.
+          term = "aqua";
+	endif
       endif
 
       ## If no 'h' (why not?) then open the terminal as Figure 0.
